ocaml/typing/env.mli

```diff
@@ -407,14 +407,14 @@ val add_signature_lazy: Subst.Lazy.signature_item list -> t -> t
    Used to implement open. Returns None if the path refers to a functor,
    not a structure. *)
 val open_signature:
-    ?used_slot:bool ref ->
-    ?loc:Location.t -> ?toplevel:bool ->
-    Asttypes.override_flag -> Path.t ->
-    t -> (t, [`Not_found | `Functor]) result
+    used_slot:bool ref ->
+    loc:Location.t -> toplevel:bool ->
+    Asttypes.override_flag -> Longident.t Location.loc ->
+    t -> Path.t * t
 
-val open_pers_signature: string -> t -> (t, [`Not_found]) result
+val open_pers_signature: loc:Location.t -> Longident.t -> t -> Path.t * t
```

Comment:
This should probably still take a `string` since e.g. `Ldot` cannot be a persistent identifier.